contracts/test/integration/IntegrationTestMakerRouter.t.sol

```diff
@@ -0,0 +1,351 @@
+            Constants.BASIS_POINTS_GRANULARITY;
+
+        assertEq((minDaiAmountOut - usdcAmount), dai.balanceOf(address(this)));
+        assertEq(
+            (usdcAmount / USDC_SCALING_FACTOR),
+            usdc.balanceOf(address(this))
+        );
+    }
+
+    function testSwapFeiForUsdcAndDaiPCVController(
+        uint64 amountFeiIn,
+        uint16 ratioUSDC
+    ) public {
+        vm.assume(amountFeiIn > 1e18);
+        vm.assume(ratioUSDC < 10_000 && ratioUSDC > 0);
+
+        vm.prank(MainnetAddresses.GOVERNOR);
+        core.grantPCVController(address(this));
+
+        uint256 minDaiAmountOut = feiPSM.getRedeemAmountOut(amountFeiIn);
+        makerRouter.swapFeiForUsdcAndDai(
+            amountFeiIn,
+            minDaiAmountOut,
+            address(this),
+            address(this),
+            ratioUSDC
+        );
+
+        uint256 usdcAmount = (minDaiAmountOut * ratioUSDC) /
+            Constants.BASIS_POINTS_GRANULARITY;
+
+        assertEq((minDaiAmountOut - usdcAmount), dai.balanceOf(address(this)));
+        assertEq(
+            (usdcAmount / USDC_SCALING_FACTOR),
+            usdc.balanceOf(address(this))
+        );
+    }
+
+    function testSwapAllFeiForDai() public {
+        vm.prank(MainnetAddresses.GOVERNOR);
+        core.grantGovernor(address(this));
+
+        makerRouter.swapAllFeiForDai(address(this));
+
+        assertEq(
+            dai.balanceOf(address(this)),
+            feiPSM.getRedeemAmountOut(mintAmount)
+        );
+    }
+
+    function testSwapAllFeiForDaiPCVController() public {
+        vm.prank(MainnetAddresses.GOVERNOR);
+        core.grantPCVController(address(this));
+
+        makerRouter.swapAllFeiForDai(address(this));
+
+        assertEq(
+            dai.balanceOf(address(this)),
+            feiPSM.getRedeemAmountOut(mintAmount)
+        );
+    }
+
+    function testSwapAllFeiForUsdc() public {
+        vm.prank(MainnetAddresses.GOVERNOR);
+        core.grantGovernor(address(this));
+        makerRouter.swapAllFeiForUsdc(address(this));
+
+        assertEq(
+            usdc.balanceOf(address(this)),
+            feiPSM.getRedeemAmountOut(mintAmount) / USDC_SCALING_FACTOR
+        );
+    }
+
+    function testSwapAllFeiForUsdcPCVController() public {
+        vm.prank(MainnetAddresses.GOVERNOR);
+        core.grantPCVController(address(this));
+        makerRouter.swapAllFeiForUsdc(address(this));
+
+        assertEq(
+            usdc.balanceOf(address(this)),
+            feiPSM.getRedeemAmountOut(mintAmount) / USDC_SCALING_FACTOR
+        );
+    }
+
+    function testSwapAllFeiForUsdcAndDai(uint16 ratioUSDC) public {
+        vm.assume(ratioUSDC < 10_000 && ratioUSDC > 0);
+
+        vm.prank(MainnetAddresses.GOVERNOR);
+        core.grantGovernor(address(this));
+
+        uint256 minDaiAmountOut = feiPSM.getRedeemAmountOut(mintAmount);
+
+        makerRouter.swapAllFeiForUsdcAndDai(
+            address(this),
+            address(this),
```

Comment:
would be great to add another test where USDC and DAI have separate destinations.